source/wp-content/plugins/wporg-developer-blog-cpts/wporg-developer-blog-cpts.php

```diff
@@ -0,0 +1,186 @@
+						],
+					],
+					[ 'core/code' ],
+					[
+						'core/heading',
+						[
+							'level'   => 2,
+							'content' => 'Relevant links',
+						],
+					],
+					[ 'core/list' ],
+					[
+						'core/paragraph',
+						[
+							'align'   => 'right',
+							'content' => '<em>Please add props here</em>',
+						],
+					],
+				],
+			],
+			'dev-blog-videos' => [
+				'labels'       => [
+					'name'          => _x( 'Videos', 'Post type general name', 'wporg' ),
+					'singular_name' => _x( 'Video', 'Post type singular name', 'wporg' ),
+					'add_new'       => __( 'Add new Video', 'wporg' ),
+				],
+				'taxonomies'   => [ 'category' ],
+				'rewrite'      => [ 'slug' => 'videos' ],
+				'public'       => true,
+				'has_archive'  => true,
+				'hierarchical' => false,
+				'show_in_rest' => true,
+				'menu_icon'    => 'dashicons-format-video',
+				'supports'     => [ 'title', 'editor', 'author', 'thumbnail' ],
+				'template'     => [
+					[
+						'core/embed',
+						[
+							'providerNameSlug' => 'youtube',
+							'responsive'       => true,
+						],
+					],
+					[
+						'core/heading',
+						[
+							'level'   => 2,
+							'content' => 'Relevant links',
+						],
+					],
+					[ 'core/list' ],
+					[
+						'core/heading',
+						[
+							'level'   => 2,
+							'content' => 'Transcript',
+						],
+					],
+					[
+						'core/paragraph',
+						[
+							'placeholder' => 'Add transcript',
+						],
+					],
+					[
+						'core/paragraph',
+						[
+							'align'   => 'right',
+							'content' => '<em>Please add props here</em>',
+						],
+					],
+				],
+			],
+		];
+
+		// load the post types
+		foreach ( $custom_post_types as $post_type => $config ) {
+			register_post_type( $post_type, $config );
+		}
+
+		// Register the coding lang taxonomy
+		$lang_tax_args = [
+			'public'       => true,
+			'labels'       => [
+				'name'          => _x( 'Coding languages', 'taxonomy general name', 'wporg' ),
+				'singular_name' => _x( 'Coding language', 'taxonomy singular name', 'wporg' ),
+				'add_new_item'  => __( 'Add New Coding language', 'wporg' ),
+				'not_found'     => __( 'No Coding languages found', 'wporg' ),
+			],
+			'show_in_rest' => true,
+		];
+		register_taxonomy( 'coding-lang', 'snippets', $lang_tax_args );
+
+		// Register the API taxonomy
+		$api_tax_args = [
+			'public'       => true,
+			'labels'       => [
+				'name'          => _x( 'APIs', 'taxonomy general name', 'wporg' ),
+				'singular_name' => _x( 'API', 'taxonomy singular name', 'wporg' ),
+				'add_new_item'  => __( 'Add New API', 'wporg' ),
+				'not_found'     => __( 'No APIs found', 'wporg' ),
+			],
+			'show_in_rest' => true,
+		];
+		register_taxonomy( 'api', 'snippets', $api_tax_args );
+	}
+);
+
+/*
+ * Display videos and snippets in the query loop block on the homepage.
+ */
+add_action(
+	'query_loop_block_query_vars',
+	function( $query ) {
+		if ( ! is_admin() && is_front_page() ) {
```

Comment:
We'd also want to include `is_home()` to account for the blog/all-posts page: https://developer.wordpress.org/news/all-posts/

```php
if ( ! is_admin() && ( is_front_page() || is_home() ) ) {
```